src/Config/MergeConfig.cs

```diff
@@ -0,0 +1,126 @@
+
+using System.Buffers;
+using System.Text;
+using System.Text.Json;
+
+namespace Azure.DataApiBuilder.Config
+{
+    /// <summary>
+    /// This Class contains methods to merge config files based on DAB_ENVIRONMENT value
+    /// to give a similar experience to that of dotnet development.
+    /// </summary>
+    public class MergeConfig
+    {
+        /// <summary>
+        /// Parse the string and call the helper methods to merge based on their value kind, i.e. Objects or Array.
+        /// </summary>
+        public static string Merge(string originalJson, string overridingJson)
+        {
+            ArrayBufferWriter<byte> outputBuffer = new();
+
+            using (JsonDocument originalJsonDoc = JsonDocument.Parse(originalJson))
+            using (JsonDocument overridingJsonDoc = JsonDocument.Parse(overridingJson))
+            using (Utf8JsonWriter jsonWriter = new(outputBuffer, new JsonWriterOptions { Indented = true }))
+            {
+                JsonElement root1 = originalJsonDoc.RootElement;
+                JsonElement root2 = overridingJsonDoc.RootElement;
+
+                if (root1.ValueKind != JsonValueKind.Array && root1.ValueKind != JsonValueKind.Object)
+                {
+                    throw new InvalidOperationException($"The original JSON document to merge new content into must be a container type. Instead it is {root1.ValueKind}.");
+                }
+
+                if (root1.ValueKind != root2.ValueKind)
+                {
+                    return originalJson;
+                }
+
+                if (root1.ValueKind == JsonValueKind.Array)
+                {
+                    MergeArrays(jsonWriter, root2);
+                }
+                else
+                {
+                    MergeObjects(jsonWriter, root1, root2);
+                }
+            }
+
+            return Encoding.UTF8.GetString(outputBuffer.WrittenSpan);
+        }
+
+        /// <summary>
+        /// This helper methods tries to merge two Json Object recursively.
+        /// </summary>
+        private static void MergeObjects(Utf8JsonWriter jsonWriter, JsonElement root1, JsonElement root2)
+        {
+            jsonWriter.WriteStartObject();
+
+            // If a property exists in both documents, either:
+            // * Merge them, if the value kinds match (e.g. both are objects or arrays) recursively,
+            // * Completely override the value of the first with the one from the second, if the value kind mismatches (e.g. one is object, while the other is an array or string),
+            // * Or favor the value of the first, if the second one is null.
+            foreach (JsonProperty property in root1.EnumerateObject())
+            {
+                string propertyName = property.Name;
+
+                JsonValueKind newValueKind;
+
+                if (root2.TryGetProperty(propertyName, out JsonElement newValue) && (newValueKind = newValue.ValueKind) != JsonValueKind.Null)
```

Comment:
is it case sensitive - TryGetProperty?

Reply:
It is Case sensitive. our properties like `Entities` are also case sensitive, it's just the values are case-insensitive.